test/transaction_changes_test.rb

```diff
@@ -126,4 +126,30 @@ def test_serialized_attributes_mutation
     @user.save!
     assert_equal '2.2.2.2', @user.connection_details.first.client_ip
   end
+
+  def test_multiple_changes_back_to_original_value
+    @user.transaction do
+      @user.name = "Dillon"
+      @user.save!(touch: false)
+
+      @user.name = "Daring"
+      @user.save!(touch: false)
+
+      @user.name = "Dylan"
+      @user.save!(touch: false)
+    end
+
+    assert_empty @user.stored_transaction_changes
+  end
+
+  def test_multiple_changes_back_to_original_value_for_serialized_attribute
+    @user.favourite_foods = ["Rice", "Sushi"]
+    @user.save!
+    @user.stored_transaction_changes = nil
+
+    @user.favourite_foods = ["Sushi", "Rice"] # sorted back in before_save
+    @user.save!(touch: false)
+
+    assert_empty @user.transaction_changed_attributes
```

Comment:
transaction_changed_attributes is meant to be used in the after_commit hook, so the tests store that in `stored_transaction_changes` to test for outside of the after_commit hook.

```suggestion
    assert_empty @user.stored_transaction_changes
```